### PR TITLE
Issue 21/cache refresh

### PR DESCRIPTION
**Pull request by tcnichol**
refresh cache script

### CODE REVIEW

**Comment by tcnichol** [comment]
This refresh script should work by itself now. It is ready for testing/review.

**Comment by max-zilla** [comment]
I'm pushing a few updates. Going forward, please remove test files/print statements before creating a PR because we don't need to keep those in production. removing some now.

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,8 @@ RUN apt-get update -qq \
       && rm -rf /srv/shiny-server/sample-apps
       
 COPY . /srv/shiny-server/
-COPY shiny-server.conf /etc/shiny-server/
```
Comment (max-zilla) [inline Dockerfile]: I'm adding this back in - we can't remove this or it breaks the app.

File: cache-refresh.R [hunk 2]
```diff
@@ -121,7 +118,12 @@ get_data_for_subexp <- function(subexp, exp_name) {
   
   # save data for given subexp
   full_cache_data[[ exp_name ]][[ subexp[['name']] ]] <- subexp_data
-  save(full_cache_data, file = "cache.RData", compress = FALSE)
+  print("saving new data to file")
+  file.create("/srv/shiny-server/cache/cache.RData.temp")
+  save(full_cache_data, file = "/srv/shiny-server/cache/cache.RData.temp", compress = FALSE)
+  # copy file to cache.RData
+  file.copy("/srv/shiny-server/cache/cache.RData.temp","/srv/shiny-server/cache/cache.RData",overwrite=TRUE)
```
Comment (max-zilla) [inline cache-refresh.R]: changing this to file.rename, we dont need to keep the temp copy for any reason

**Comment by max-zilla** [comment]
We need to make sure that when cache-server reloads the cache file, that the app.R script correctly uses the new file instead of using the already-loaded data.

**Comment by tcnichol** [comment]
@robkooper 

i refactored the loading of the cache data and added a check for whether the cache file has been modified. not sure if it's correct, so please check and comment.

**Pull request merged**